storage/types.go

```diff
@@ -420,11 +418,12 @@ func (n *NodeID) Split(prefixBytes, suffixBits int) ([]byte, *Suffix) {
 	suffixBytes := bytesForBits(b)
 	maskIndex := (b - 1) / 8
 	maskLowBits := (b-1)%8 + 1
-	sfxPath := a[prefixBytes : prefixBytes+suffixBytes]
+	sfxPath := make([]byte, suffixBytes)
+	copy(sfxPath, n.Path[prefixBytes:prefixBytes+suffixBytes])
 	sfxPath[maskIndex] &= ((0x01 << uint(maskLowBits)) - 1) << uint(8-maskLowBits)
 	sfx := NewSuffix(byte(b), sfxPath)
 
-	return a[:prefixBytes], sfx
+	return n.Path[:prefixBytes], sfx
```

Comment:
Slightly worried about that one. Some of the NodeID methods are mutators. If they share path slices this could break.

Reply:
Fair point - I've reverted this, and factored out the suffix creation bit into its own method and switched over to that in the places where we discard the prefix anyway.

Reply:
Defensive copying LGTM.